diff --git a/debian/lib/python/debian_linux/abi.py b/debian/lib/python/debian_linux/abi.py
index c3ca9704045cd16c345034d0d29b032c27b188cd..8c108b6fdfbf410e8373c2ba30ecb640d5b5bbbd 100644
--- a/debian/lib/python/debian_linux/abi.py
+++ b/debian/lib/python/debian_linux/abi.py
@@ -35,10 +35,10 @@ class Symbols(dict):
 
     def read(self, file):
         for line in file:
-            version, name, namespace, module, export = line.strip().split('\t')
+            version, name, module, export, namespace = line.strip('\r\n').split('\t')
             self[name] = Symbol(name, namespace, module, version, export)
 
     def write(self, file):
         for s in sorted(self.values(), key=lambda i: i.name):
             file.write("%s\t%s\t%s\t%s\t%s\n" %
-                       (s.version, s.name, s.namespace, s.module, s.export))
+                       (s.version, s.name, s.module, s.export, s.namespace))
