diff --git a/debian/config/amd64/config b/debian/config/amd64/config
index 4a4e34aac1ce5eb6a45e25bb9d7f7b227d479564..99a81253d3b6b813b5d0956e3927698d723dbcb9 100644
--- a/debian/config/amd64/config
+++ b/debian/config/amd64/config
@@ -257,6 +257,13 @@ CONFIG_ZONE_DEVICE=y
 ##
 CONFIG_LSM_MMAP_MIN_ADDR=65536
 
+##
+## file: sound/soc/amd/Kconfig
+##
+CONFIG_SND_SOC_AMD_ACP3x=m
+CONFIG_SND_SOC_AMD_RENOIR=m
+CONFIG_SND_SOC_AMD_RENOIR_MACH=m
+
 ##
 ## file: sound/soc/sof/intel/Kconfig
 ##
