diff --git a/aarch64.cfg b/aarch64.cfg
index e69de29b..f5afeb81 100644
--- a/aarch64.cfg
+++ b/aarch64.cfg
@@ -0,0 +1,23 @@
+{
+	"name": "qemu-aarch64",
+	"target": "linux/arm64",
+	"http": "localhost:12346",
+	"rpc": "127.0.0.1:5555",
+	"workdir": "./workdir64",
+	"kernel_obj": "../../../../../buildroot/output/build/linux-5.9.12/vmlinux",
+	"kernel_src": "../../../../../buildroot/output/build/linux-5.9.12",
+	"syzkaller": ".",
+	"image": "../../../../../buildroot/output/images/rootfs.ext2",
+	"sandbox": "namespace",
+	"reproduce": false,
+	"procs": 8,
+	"type": "qemu",
+	"vm": {
+		"count": 1,
+		"cmdline": "console=ttyAMA0,115200 root=/dev/vda",
+		"kernel": "../../../../../buildroot/output/images/Image",
+		"image_device": "vd",
+		"mem": 1024,
+		"cpu": 4
+	}
+}
diff --git a/arm.cfg b/arm.cfg
index e69de29b..a6384142 100644
--- a/arm.cfg
+++ b/arm.cfg
@@ -0,0 +1,24 @@
+{
+	"name": "qemu-arm",
+	"target": "linux/arm",
+	"http": "localhost:12345",
+	"rpc": "127.0.0.1:5556",
+	"workdir": "./workdir",
+	"kernel_obj": "../../../../../buildroot/output/build/linux-5.9.12/vmlinux",
+	"kernel_src": "../../../../../buildroot/output/build/linux-5.9.12",
+	"syzkaller": ".",
+	"image": "../../../../../buildroot/output/images/rootfs.ext2",
+	"sandbox": "namespace",
+	"reproduce": false,
+	"procs": 8,
+	"type": "qemu",
+	"vm": {
+		"count": 1,
+		"qemu_args": "-M vexpress-a9 -cpu cortex-a9 -dtb ../../../../../buildroot/output/images/vexpress-v2p-ca9.dtb",
+		"cmdline": "console=ttyAMA0,115200 root=/dev/mmcblk0",
+		"kernel": "../../../../../buildroot/output/images/zImage",
+		"image_device": "sd",
+		"mem": 1024,
+		"cpu": 4
+	}
+}
diff --git a/sys/targets/targets.go b/sys/targets/targets.go
index 3299e52b..1b8b045e 100644
--- a/sys/targets/targets.go
+++ b/sys/targets/targets.go
@@ -179,8 +179,8 @@
 			PtrSize:          4,
 			PageSize:         4 << 10,
 			LittleEndian:     true,
-			CFlags:           []string{"-D__LINUX_ARM_ARCH__=6", "-march=armv6"},
-			Triple:           "arm-linux-gnueabi",
+			CFlags:           []string{"-D__LINUX_ARM_ARCH__=7", "-march=armv7-a"},
+			Triple:           "arm-linux-gnueabihf",
 			KernelArch:       "arm",
 			KernelHeaderArch: "arm",
 		},
diff --git a/vm/qemu/qemu.go b/vm/qemu/qemu.go
index dae8f493..6ce6050c 100644
--- a/vm/qemu/qemu.go
+++ b/vm/qemu/qemu.go
@@ -134,6 +134,7 @@
 		QemuArgs:  "-machine virt,virtualization=on -cpu cortex-a57",
 		TargetDir: "/",
 		NetDev:    "virtio-net-pci",
+		UseNewQemuImageOptions: true,
 		CmdLine: append(linuxCmdline,
 			"root=/dev/vda",
 			"console=ttyAMA0",
--- a/executor/common_linux.h
+++ b/executor/common_linux.h
@@ -2004,7 +2004,9 @@
 #if !GOARCH_riscv64
 #include <errno.h>
 #include <fcntl.h>
+#if !GOARCH_arm
 #include <linux/kvm.h>
+#endif
 #include <stdarg.h>
 #include <stddef.h>
 #include <sys/ioctl.h>
